src/System.Private.CoreLib/src/System/AppContext/AppContext.cs

```diff
@@ -7,29 +7,20 @@
 using System.IO;
 using System.Reflection;
 using System.Runtime.Versioning;
+using System.Threading;
 
 namespace System
 {
     public static class AppContext
     {
-        [Flags]
-        private enum SwitchValueState
-        {
-            HasFalseValue = 0x1,
-            HasTrueValue = 0x2,
-            HasLookedForOverride = 0x4,
-            UnknownValue = 0x8 // Has no default and could not find an override
-        }
-        private static readonly Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
+        private static Dictionary<string, object> s_localStore = new Dictionary<string, object>();
+        private static Dictionary<string, int> s_switches;
```

Comment:
Why store ints if we only ever use them as 1/-1 and expose them as booleans?

Reply:
Good point...